File: src/main/java/com/kamikazejam/datastore/util/PlayerUtil.java
package com.kamikazejam.datastore.util;

import org.bukkit.entity.Player;
import org.jetbrains.annotations.Nullable;

/**
 * Utility class for helping with players
 * You can give items (which drop near them if they are full), and clean their inventory (with or without armor)
 */
@SuppressWarnings("unused")
public class PlayerUtil {
    /**
     * @return true IFF (player != null AND player.isOnline() AND player.isValid())
     */
    public static boolean isFullyValidPlayer(@Nullable Player player) {
        return player != null && player.isOnline() && player.isValid();
    }
}

File: src/main/kotlin/com/kamikazejam/datastore/util/PlayerUtil.kt
package com.kamikazejam.datastore.util

import org.bukkit.entity.Player

/**
 * Utility class for helping with players
 * You can give items (which drop near them if they are full), and clean their inventory (with or without armor)
 */
@Suppress("unused")
object PlayerUtil {
    /**
     * @return true IFF (player != null AND player.isOnline() AND player.isValid())
     */
    fun isFullyValidPlayer(player: Player?): Boolean {
        return player != null && player.isOnline && player.isValid
    }
}
